### ISSUE TITLE
gammainc handles inf incorrectly

### ISSUE THREAD

**Issue by terhorst** [google/jax#6535]
```python
import scipy.special
import jax.scipy.special
scipy.special.gammainc(1, float('inf'))
# 1.0
jax.scipy.special.gammainc(1, float('inf'))
# DeviceArray(nan, dtype=float32)
```

**Comment by terhorst** [comment]
`gammainc` just wraps `lax.igamma` so maybe it's best to ask for a resolution upstream?

**Comment by hawkinsp** [comment]
Ultimately the code JAX is calling is here:
https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/compiler/xla/client/lib/math.cc;l=926

It's probably best and simplest if we update that code. I'm not an expert on this function: is the fix as simple as testing for an `inf` input?

**Comment by terhorst** [comment]
`gammainc(a, inf)` should equal 1 at least for all `a > 0`. It is also defined for on the whole complex plane excepting the non-positive integers (see "[Limit for z → +∞](https://en.wikipedia.org/wiki/Incomplete_gamma_function)"), but scipy returns `NaN` for `x < 0` [1], and throws `TypeError` if `x` is complex. So an acceptable test is probably:
```
def igamma(a, x):
  if isinf(x):
    if a>0: return 1.
    return float('nan')
  ...
```

[1] Actually `scipy.special.gammainc(0., inf) == 1.` as well, but I think this is wrong.

**Comment by OE-GOD** [comment]
This bug still reproduces at `jax 0.10.1` (this issue was closed 2021):

```python
import jax
jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jss
import scipy.special as ss
import numpy as np

# Original reproducer from this issue:
print(ss.gammainc(1, np.inf))                                  # 1.0
print(float(jss.gammainc(np.float32(1), np.float32(np.inf))))  # nan  ← still broken

# Other inputs that surface the same / a related issue at jax 0.10.1:
print(ss.gammainc(np.inf, np.inf), float(jss.gammainc(np.float32(np.inf), np.float32(np.inf))))   # scipy=nan jax=1.0   ← inverted
print(ss.gammainc(0.0, 0.5),       float(jss.gammainc(np.float32(0.0), np.float32(0.5))))         # scipy=1.0 jax=nan   ← still broken
```

The boundary-behaviour class of bugs around `gammainc/gammaincc` is also tracked in #20507 (also CLOSED, also still reproduces). Re-opening this issue (or folding into #20507) might be appropriate.

**Comment by jakevdp** [comment]
Hi @OE-GOD – can you say more about the environment in which you ran this? I tried executing your code snippet at `print(float(jss.gammainc(np.float32(1), np.float32(np.inf))))` printed `1.0`.

**Comment by OE-GOD** [comment]
@jakevdp — apologies, you're right. I re-ran the original reproducer carefully on my own machine and it returns `1.0` correctly now, not `nan`. The original bug from this 2021 report **is fixed**. I retract my "still reproduces" claim for that specific input.

What actually still reproduces at `jax 0.10.1`, `jaxlib 0.10.1`, `scipy 1.17.1`, Python 3.12.13, macOS arm64 (CPU backend), `jax_enable_x64=True`:

```python
import jax
jax.config.update('jax_enable_x64', True)
import jax.scipy.special as jss
import scipy.special as ss
import numpy as np

# Original case from this issue (now FIXED, returning 1.0):
print(float(jss.gammainc(np.float32(1), np.float32(np.inf))))                     # 1.0 ✓

# Two cases that DO still disagree with scipy:
print(ss.gammainc(0.0, 0.5),   float(jss.gammainc(np.float32(0.0), np.float32(0.5))))   # scipy=1.0 jax=nan
print(ss.gammainc(np.inf, np.inf), float(jss.gammainc(np.float32(np.inf), np.float32(np.inf))))  # scipy=nan jax=1.0
```

These two are different from the original report — they're boundary cases at `(0, x)` and `(inf, inf)` rather than `(1, inf)`. They belong on #20507 (boundary behaviour of `gammainc`/`gammaincc`) rather than re-opening this one. I'll move that observation to #20507.

Sorry for the noise on this issue — I conflated my recent finding with the original reproducer when I posted. The original fix appears to have held.
